### PR TITLE
collections: fix some Counter nits

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

**Comment by hauntsaninja** [comment]
Okay, looks like the positional-only stuff was changed in https://github.com/python/cpython/pull/20548

File: stdlib/collections/__init__.pyi [hunk 1]
```diff
@@ -277,9 +277,15 @@ class Counter(dict[_T, int], Generic[_T]):
...
+    def __missing__(self, key: _T) -> int: ...
     def __delitem__(self, elem: object) -> None: ...
-    def __eq__(self, other: object) -> bool: ...
-    def __ne__(self, other: object) -> bool: ...
+    if sys.version_info >= (3, 10):
+        def __eq__(self, other: object) -> bool: ...
+        def __ne__(self, other: object) -> bool: ...
+    else:
+        def __eq__(self, __other: object) -> bool: ...
+        def __ne__(self, __other: object) -> bool: ...
```
Comment (Akuli) [inline stdlib/collections/__init__.pyi:287]: These don't seem to exist on 3.9, 3.8, 3.7 or 3.6.

Comment (hauntsaninja) [reply stdlib/collections/__init__.pyi:287]: You're right, it inherits them from `__dict__`.

Comment (hauntsaninja) [inline stdlib/collections/__init__.pyi:287]:
```suggestion

```

**Review by Akuli — APPROVED** [review]
Please merge when CI passes.

**Pull request merged**